File: src/main/kotlin/com/stremio/core/models/Link.kt
package com.stremio.core.models

import com.stremio.core.models.common.Loadable
import com.stremio.core.types.api.LinkCodeResponse

data class Link<T>(
    val code: Loadable<LinkCodeResponse>?,
    val data: Loadable<T>?
)

File: src/main/rust/model/fields/link.rs
use stremio_core::models::link::Link;
use stremio_core::types::api::LinkAuthKey;

use crate::bridge::{ToProtobuf, ToProtobufAny};
use crate::protobuf::stremio::core::models;

impl ToProtobuf<models::AuthLink, ()> for Link<LinkAuthKey> {
    fn to_protobuf(&self, _args: &()) -> models::AuthLink {
        models::AuthLink {
            code: self.code.to_protobuf(&()),
            data: self.data.to_protobuf(&()),
        }
    }
}
